### ISSUE TITLE
Dependency Track API fails to start in Fargate Task

### ISSUE THREAD

**Issue by mikehall-mozz** [DependencyTrack/dependency-track#3824]
### Current Behavior

[log-events-viewer-result.csv](https://github.com/user-attachments/files/15742186/log-events-viewer-result.csv)

Hi there, I am trying to run Dependency Track in a Fargate Task on AWS, and it hangs during startup. I have enabled debug logging, and it doesn't give me much to go on. The last line I see is as follows:

2024-06-07T15:59:39.280Z	2024-06-07 15:59:39,279 INFO [PersistenceManagerFactory] Creating transactional connection pool

There are no further lines logged, and the task is eventually killed as it fails health checks.

I can see that the user has sufficient permissions in the database, because it creates the tables successfully.

### Steps to Reproduce

Run dependency track in a Fargate task with the following env vars:
nvironment : [
        {
          name : "ALPINE_DATABASE_MODE",
          value : "external"
        },
        {
          name : "ALPINE_DATABASE_URL",
          value : "jdbc:postgresql://\<database URL\>"
        },
        {
          name : "ALPINE_DATABASE_USERNAME",
          value : "\<user\>"
        },
        {
          name : "ALPINE_DATABASE_PASSWORD",
          value : "\<password\>"
        },
        {
          name : "ALPINE_DATABASE_POOL_ENABLED",
          value : "true"
        },
        {
          name : "ALPINE_DATABASE_POOL_MAX_SIZE",
          value : "20"
        },
        {
          name : "ALPINE_DATABASE_POOL_MIN_IDLE",
          value : "10"
        },
        {
          name : "ALPINE_DATABASE_POOL_IDLE_TIMEOUT",
          value : "300000"
        },
        {
          name : "ALPINE_DATABASE_POOL_MAX_LIFETIME",
          value : "600000"
        },
        {
          name: "LOGGING_LEVEL",
          value: "DEBUG,"
        },
        {
          name : "region",
          value : "${var.region}"
        },
        {
          name : "env",
          value : "${var.environment}"
        }
      ]

### Expected Behavior

Dependency Track API starts successfully

### Dependency-Track Version

4.11.3

### Dependency-Track Distribution

Container Image

### Database Server

PostgreSQL

### Database Server Version

16.1

### Browser

Google Chrome

### Checklist

- [X] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/dependency-track/blob/master/CONTRIBUTING.md#filing-issues)
- [X] I have checked the [existing issues](https://github.com/DependencyTrack/dependency-track/issues) for whether this defect was already reported

**Comment by nscuro** [comment]
> ```
> {
> name : "ALPINE_DATABASE_URL",
> value : "jdbc:postgresql://"
> },
> {
> name : "ALPINE_DATABASE_USERNAME",
> value : ""
> },
> {
> name : "ALPINE_DATABASE_PASSWORD",
> value : ""
> },
> ```

Did you redact the values or is that how the task is configured?

If the logs stop at:

```
2024-06-07T15:59:39.280Z 2024-06-07 15:59:39,279 INFO [PersistenceManagerFactory] Creating transactional connection pool
```

then that's a hint that `ALPINE_DATABASE_URL` points to a host that can't be reached.

**Comment by mikehall-mozz** [comment]
Yes @nscuro , not sure why they have ended up blank, they contained <redacted values> :)

Definitely confirmed connection working with another user that didn't have appropriate permissions, who was getting rejected with the unable to create DELETEMEXXXXXX (or whatever it's called) table.

**Comment by mikehall-mozz** [comment]
So I have managed to get it working by setting ALPINE_DATABASE_POOL_ENABLED to "false", so it seems like it is failing to set up the pool correctly. Any ideas as to how to get that working @nscuro ?

**Comment by nscuro** [comment]
You could try reducing the connection pool size to increasingly smaller numbers until it works. I've not seen the startup being blocked by connection creation before, so not sure what might be the issue.

Another thing you could investigate is whether your database has a connection limit that prevents DT from creating more than a handful of connections.

**Comment by sbkg0002** [comment]
@mikehall-mozz did you successfully run Dendency Track in Fargate? Can you share details on what you use?

**Comment by WIStudent** [comment]
I had the same issue, I forgot to set `ALPINE_DATABASE_DRIVER` to `org.postgresql.Driver`. The API server started normally on fargate after I added it.
